tests/ha/control_pod/test_control_pod_softfailure.py

```diff
@@ -279,3 +280,166 @@ def test_45498(self):
 
         LOGGER.info("ENDED: Verify IAM user creation/deletion and IOs on already created IAM user "
                     "while performing control pod soft-failure in loop 1 by 1 for N control pods.")
+
+    @pytest.mark.ha
+    @pytest.mark.lc
+    @pytest.mark.tags("TEST-45501")
+    def test_45501(self):
+        """
+        Verify IAM user creation and IOs before and after soft-failures
+        """
+        LOGGER.info("STARTED: Verify IAM user creation & IOs before and after soft-failures.")
+
+        LOGGER.info("Step 1: Perform IOs before soft-failure.")
+        self.test_prefix = 'test-45501'
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(self.s3_clean.values())[0],
+                                                    log_prefix=self.test_prefix, skipcleanup=True,
+                                                    nsamples=2, nclients=2)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 1: Performed IOs before soft-failure.")
+
+        LOGGER.info("Step 2: Perform soft-failure of control pod and check cluster status.")
+        pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.CONTROL_POD_NAME_PREFIX)
+        pod_name = self.system_random.sample(pod_list, 1)[0]
+        self.node_master_list[0].kill_process_in_container(pod_name=pod_name,
+                                                           container_name=const.CORTX_CSM_POD,
+                                                           process_name=const.CSM_AGENT_PRC,
+                                                           safe_kill=True)
+        LOGGER.info("Step 2.1: Check cluster has failures.")
+        resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+        assert_utils.assert_false(resp[0], resp)
```

Comment:
are we sure without delay as well we will get cluster failures? Or rather can we rather wait for 30s which is container restart time and then check for *online* status?